ui/pages/asset/components/asset-header.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { forwardRef, useImperativeHandle, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import {
+  AlignItems,
+  BackgroundColor,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import {
+  AvatarToken,
+  AvatarTokenSize,
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+  Text,
+} from '../../../components/component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { DEFAULT_ROUTE } from '../../../helpers/constants/routes';
+
+const AssetHeader = forwardRef(
+  (
+    {
+      image,
+      balance,
+      optionsButton,
+    }: {
+      image: string;
+      balance: {
+        display: string;
+        fiat?: string;
+      };
+      optionsButton: React.ReactNode;
+    },
+    ref,
+  ) => {
+    const t = useI18nContext();
+    const history = useHistory();
+
+    const [balanceOpacity, setBalanceOpacity] = useState(0);
+    useImperativeHandle(ref, () => ({ setBalanceOpacity }));
+
+    return (
+      <Box
+        display={Display.Flex}
+        backgroundColor={BackgroundColor.backgroundDefault}
+        justifyContent={JustifyContent.spaceBetween}
+        flexDirection={FlexDirection.Row}
+        alignItems={AlignItems.center}
+        padding={4}
+        className={'asset-header'}
+      >
+        <ButtonIcon
+          size={ButtonIconSize.Sm}
+          ariaLabel={t('back')}
+          iconName={IconName.ArrowLeft}
+          onClick={() => history.push(DEFAULT_ROUTE)}
+        />
+        {balanceOpacity > 0 && (
+          <Box
+            style={{ opacity: balanceOpacity }}
+            display={Display.Flex}
+            flexDirection={FlexDirection.Column}
+            alignItems={AlignItems.center}
+          >
+            <Text variant={TextVariant.bodyMdBold}>{balance.fiat}</Text>
+            <Box display={Display.Flex} alignItems={AlignItems.center} gap={2}>
+              <AvatarToken src={image} size={AvatarTokenSize.Xs} />
+              <Text
+                variant={TextVariant.bodySm}
+                color={TextColor.textAlternative}
+              >
+                {balance.display}
+              </Text>
+            </Box>
+          </Box>
+        )}
+        {optionsButton}
+      </Box>
+    );
+  },
+);
+
```

Comment:
you can add props validation here:
```
AssetHeader.propTypes = {
  image: PropTypes.string.isRequired,
  balance: PropTypes.shape({
    display: PropTypes.string.isRequired,
    fiat: PropTypes.string,
  }).isRequired,
  optionsButton: PropTypes.node.isRequired,
};
```

Reply:
I thought this wasn't needed for newer `.tsx` components since they get types from typescript.  But I'm not 100% sure.